core/app/models/spree/variant.rb

```diff
@@ -60,7 +60,7 @@ class Variant < Spree::Base
 
     after_touch :clear_in_stock_cache
 
-    scope :in_stock, -> { joins(:stock_items).where('count_on_hand > ? OR track_inventory = ?', 0, false) }
+    scope :in_stock, -> { joins(:stock_items).where("#{Spree::StockItem.table_name}.count_on_hand > ? OR track_inventory = ?", 0, false) }
```

Comment:
@kgorazd we should also add the table name for track_inventory attribute as well